### PR TITLE
Add SASS support

### PR DESCRIPTION
**Pull request by rafayet-monon**
Task: https://github.com/nimblehq/react-templates/projects/2#card-59675004

## What happened 

Support SASS in the template.

## Insight

- Added node-sass package to support SASS
- Added Stylelint
- Convert current CSS files to SCSS file
- Added [stylelint-config-sass-guidelines](https://github.com/bjankord/stylelint-config-sass-guidelines) and [stylelint-scss](https://github.com/kristerkari/stylelint-scss) plugin for linting stylesheets.
- Added [stylelint-order](https://github.com/hudochenkov/stylelint-order) and [stylelint-config-property-sort-order-smacss](https://github.com/cahamilton/stylelint-config-property-sort-order-smacss) to order the SCSS by [SMACSS order](https://github.com/cahamilton/css-property-sort-order-smacss/blob/v2.1.3/index.js) rule. Property groups are ordered in the following way, and each group is differentiated with a blank space for readability:
    - Heading
    - Box
    - Animation
    - Border
    - Background
    - Text
    - Other

## Proof Of Work 💪

- Test locally by running
`npx create-react-app my-react-app --template file:<path-to-the-template>`

- Run the command `yarn stylelint` after creating the project. There will be no style-related linting errors.
image

- SCSS order violation linting error
image

- SASS rule violation linting error
image

### CODE REVIEW

File: template.json [hunk 1]
```diff
@@ -21,12 +21,20 @@
...
+      "stylelint-order": "4.1.0",
+      "stylelint-config-property-sort-order-smacss": "7.1.0"
     },
     "scripts": {
       "lint": "eslint ./src --ext .ts,.tsx",
       "lint:fix": "eslint ./src --ext .ts,.tsx --fix",
-      "prettier": "prettier \"src/**/*.{ts,html}\" --write"
+      "prettier": "prettier \"src/**/*.{ts,html}\" --write",
+      "stylelint": "stylelint '**/*.scss'",
+      "stylelint:fix": "stylelint '**/*.scss' --fix"
```
Comment (carryall) [inline template.json:37]: @rafayet-monon As we have multiple linting and also prettier here How about adding a script to run all of them at once? For example
```json
    "codebase": "yarn lint && yarn prettier && yarn stylelint",
    "codebase:fix": "yarn lint:fix && yarn prettier && yarn stylelint:fix",
```

File: template/src/App.scss [hunk 2]
```diff
@@ -1,37 +1,41 @@
-.App {
+.app {
```
Comment (carryall) [inline template/src/App.scss:1]: @rafayet-monon How about moving this file to `template/src/assets/stylesheets` and also rename it to `app.scss`? 🤔

**Review by carryall — COMMENTED** [review]
@rafayet-monon As we planned to keep all styles for screens and components in `src/assets/stylesheets` how about adding `tempalte/src/assets/stylesheets/application.scss`? so we can have the folder structure for stylesheets like

```
-stylesheets
--base
--components
--pages
--layouts
```

and in `application.scss` we can import them
```scss
// Bases
@import './base/*';

// Components
@import './components/*';

// Layouts
@import './layouts/*';

// Screens
@import './screens/*';

```

**Comment by rafayet-monon** [comment]
@carryall New changes in [e55df59](https://github.com/nimblehq/react-templates/pull/20/commits/e55df5902a075161a78f982a1c97a83b5b62e2f6)

- Moved `app.scss` and `index.scss` in `template/src/assets/stylesheets`
- Added `codebase:lint` and `codebase:fix` commands
- Removed prettier commands as ESLint is configured to detect and fix prettier errors.
- Added stylesheet folder structure.

File: template/src/App.tsx [hunk 3]
```diff
@@ -1,17 +1,18 @@
 import React from 'react'
 
-import logo from './logo.svg'
-import './App.css'
+import logo from './assets/images/logo.svg'
```
Comment (noor-ahmed) [inline template/src/App.tsx:3]: Can we configure relative path for these cases?

File: template/src/App.tsx [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
+import './dummy.scss'
+import './assets/stylesheets/application.scss'
```
Comment (noor-ahmed) [inline template/src/App.tsx:5]: Ditto

Comment (rafayet-monon) [reply template/src/App.tsx:5]: @noor-ahmed I have already created an issue for this purpose - https://github.com/nimblehq/react-templates/issues/22
I was planning to configure it in a different PR. Is that okay?

Comment (noor-ahmed) [reply template/src/App.tsx:5]: Thats also nice. ✌🏻

**Review by noor-ahmed — APPROVED** [review]

**Review by carryall — APPROVED** [review]

**Pull request merged**